python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -19,10 +19,116 @@
 from polars.polars import _expr_nodes as pl_expr, _ir_nodes as pl_ir
 
 from cudf_polars.dsl import expr, ir
-from cudf_polars.typing import NodeTraverser
 from cudf_polars.utils import dtypes
 
-__all__ = ["translate_ir", "translate_named_expr"]
+if TYPE_CHECKING:
+    from cudf_polars.typing import NodeTraverser
+
+__all__ = ["Translator", "translate_named_expr"]
+
+
+class Translator:
+    """
+    Translates polars-internal IR nodes and expressions to our representation.
+
+    Parameters
+    ----------
+    visitor
+        Polars NodeTraverser object
+    debug_mode
+        Setting this mode to True allows Translator to collect
+        errors raised during translation
+    """
+
+    def __init__(self, visitor: NodeTraverser, *, debug_mode: bool = False):
+        self.visitor = visitor
+        self.debug_mode = debug_mode
+        self.errors: list[str] = []
+
+    def translate_ir(self, *, n: int | None = None) -> ir.IR:
+        """
+        Translate a polars-internal IR node to our representation.
+
+        Parameters
+        ----------
+        n
+            Optional node to start traversing from, if not provided uses
+            current polars-internal node.
+
+        Returns
+        -------
+        Translated IR object
+
+        Raises
+        ------
+        NotImplementedError
+            If the version of Polars IR is unsupported.
+
+        Notes
+        -----
+        Any expression nodes that cannot be translated are replaced by
+        :class:`expr.ErrorNode` nodes and collected in the the `errors` attribute.
+        After translation is complete, this list of errors should be inspected
+        to determine if the query is supported.
+        """
+        ctx: AbstractContextManager[None] = (
+            set_node(self.visitor, n) if n is not None else noop_context
+        )
+        # IR is versioned with major.minor, minor is bumped for backwards
+        # compatible changes (e.g. adding new nodes), major is bumped for
+        # incompatible changes (e.g. renaming nodes).
+        # Polars 1.7 changes definition of the CSV reader options schema name.
```

Comment:
```suggestion
```

This comment is now out of date, so let's just remove it, since we require polars >= 1.8